web/src/types.ts

```diff
@@ -63,17 +73,60 @@ export type RawMessage = {
     sender_realm_str: string;
     stream_id?: number;
     subject: string;
-    submessages: string[];
+    submessages: Submessage[];
```

Comment:
Nice catch! It turned out that our test cases which have been using `string[]` are not so representative of the real data.